integration-tests/utils/integration-utils.js

```diff
@@ -119,6 +123,15 @@ const waitForNoSpinners = page => {
   return page.waitForXPath('//*[@data-icon="loadingSpinner"]', { hidden: true })
 }
 
+// waitForNoSpinnersAfterAction
+// This takes advantage of the mutationObserver in puppeteer by initializing the wait before taking an action.
```

Comment:
[`MutationObserver`](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver) does look like it could be useful. This function is using `Promise.all` which may also be effective in this case. I'm still a little suspicious that there could be a gap here, but I also can't think of a case where this wouldn't work. 🤔  Anyway, I just wanted to make sure the code and comment are consistent.